EIPS/eip-2212.md

```diff
@@ -0,0 +1,271 @@
+---
+eip: 2212
+title: ERC-2212 Interest Earning Stakes
+author: Paul Razvan Berg (@PaulRBerg) <paul@sablier.finance>
+discussions-to: https://github.com/ethereum/EIPs/issues/2212
+status: Draft
+type: Standards Track
+category: ERC
+created: 2019-07-26
+requires: 20
+---
+
+## Simple Summary
+
+We propose a new way to monetise decentralised apps that involves users staking tokens in exchange for a product or a service and creators earning interest on the pooled stakes through a continuous lending protocol such as [Compound](https://compound.finance).
+
+## Abstract
+
+The advent of decentralised lending brought to the web3 ecosystem a new primitive for designing business models. We herein describe a smart contract interface that accepts deposits ("stakes") in [ERC-20](https://eips.ethereum.org/EIPS/eip-20) tokens. These deposits are immediately lent on the market to earn interest either for just the owner of the contract, or for both them and the user. We assume that users get access to a product or a service for as long as they keep staking tokens.
+
+## Motivation
+
+While most web3 business models rely on paying a percentage fee on transfers, or committing to a monthly subscription, we aim to flip the model on its head. Users are their own banks now, so, instead of charging them, we make them stake tokens that can be claimed back, in full, at any point in time. We view this as a win-win-win scenario for users, dapp creators and lending protocols.
+
+It is worth it to mention PoolTogether, a "no-loss lottery" that launched recently. It works by having participants deposit money in their contracts, locking it up for a while and earning interest through Compound. Ultimately, they choose a lucky winner and return the deposits back to all other players. PoolTogether takes a 10% commission on the prize.
+
+We wrote this specification because interest earning stakes are a powerful financial instrument and a community-vetted standard and implementation will reduce the cost of replication.
+
+### Use Cases
+
+- Software products and services
+- Charity pools
+- Games
+- Gym memberships
+
+## Specification
+
+Assigning an owner is required for this spec to make sense, thus we assume that a proper implementation would use a
+well-known contract such as OpenZeppelin's
+[Ownable](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/ownership/Ownable.sol). For
+brevity, we haven't included any method for assigning or transferring ownership.
+
+Every time we refer to cTokens, we mean Compound Tokens, as defined in the official
```

Comment:
Hm, I don't see a specification there. What is the interface of a cToken and why does ERC-2212 need to use cTokens and not any ERC-20 ?

Reply:
Afaik, Compound did not publish the interface for cTokens in this repo, only on their website.

This spec can't use ERC-20, because of the need to calculate interest and withhold a portion of it. Then, I didn't include other protocols other than Compound merely because it would've taken more time and I wanted to get it out there and get feedback.

If you read the latest messages in https://github.com/ethereum/EIPs/issues/2212, you can  see that we started working on a v2, which is protocol-agnostic. v1 is not.